e2e/LongHaul/device/IotHub.cs

```diff
@@ -198,22 +161,26 @@ public async Task RunAsync(CancellationToken ct)
             _messagesToSend.Enqueue(iotMessage);
         }
 
-        public async Task SetPropertiesAsync(object properties, CancellationToken cancellationToken)
+        public async Task SetPropertiesAsync(SystemProperties properties, CancellationToken cancellationToken)
         {
             Debug.Assert(_deviceClient != null);
             Debug.Assert(properties != null);
 
-            string propertiesPayload = JsonSerializer.Serialize(properties, _jsonSerializerOptions);
-            Debug.Assert(!string.IsNullOrWhiteSpace(propertiesPayload));
+            var reportedProperties = new ReportedProperties
+            {
+                { "systemArchitecture", properties.SystemArchitecture },
+                { "osVersion", properties.OsVersion },
+                { "frameworkDescription", properties.FrameworkDescription }
+            };
 
             await _deviceClient
                 .UpdateReportedPropertiesAsync(
-                    new TwinCollection(propertiesPayload),
+                    reportedProperties,
                     cancellationToken)
                 .ConfigureAwait(false);
         }
 
-        public void Dispose()
+        public async void Dispose()
```

Comment:
Change to `ValueTask DisposeAsync()` and update class to inherit from `IAsyncDisposable`.